src/components/implementation/resmgr/naive/res_mgr.c

```diff
@@ -0,0 +1,324 @@
+#include <cos_component.h>
+#include <cos_kernel_api.h>
+#include <cos_defkernel_api.h>
+#include <sl.h>
+#include <resmgr.h>
+#include "res_info.h"
+
+thdcap_t
+resmgr_thd_create_intern(spdid_t cur, int idx)
+{
+	struct cos_defcompinfo *res_dci = cos_defcompinfo_curr_get();
+	struct cos_compinfo *res_ci     = cos_compinfo_get(res_dci);
+	struct res_comp_info *r = res_info_comp_find(cur);
+	struct sl_thd *rt = NULL, *t = NULL;
+	int ret;
+
+	assert(r && res_info_init_check(r));
+	assert(res_info_is_sched(cur));
+	assert(idx > 0);
+
+	t = sl_thd_ext_idx_alloc(res_info_dci(r), idx);
+	assert(t);
+	rt = res_info_thd_init(r, t);
+	assert(rt);
+
+	ret = cos_cap_cpy(res_info_ci(r), res_ci, CAP_THD, sl_thd_thdcap(rt));
+	assert(ret > 0);
+
+	return ret;
+}
+
+thdcap_t
+resmgr_ext_thd_create_intern(spdid_t cur, spdid_t s, int idx)
```

Comment:
`_ext_` seems out of place in the "noun" part of the name.

`resmgr_thd_create_ext_intern`?
`resmgr_thd_create_in_intern`?  (as in create the thread *in* another component)

Reply:
I'll change to use `capmgr_thd_create_in`. I'll need to change in a lot of places (including sl, cos_defkernel_api also), will do it.